File: flowers.cpp
#include <iostream>
#include <math.h>
#include <limits>

using namespace std;
int grads = 8000000;

double f(double a, double b, double x)
{
  return a * pow(M_E, -pow(x, 2)) + b * sqrt(x);
}

double vol(double a, double b, double h)
{
  double height = h / (double)grads;

  double res = 0.0;

  for (int i = 0; i < grads; i++)
  {
    double x = i * height + (height / 2.0);
    double r = f(a, b, x);
    r = pow(r, 2);
    // if (i != 0 && i != grads - 1)
    // {
    //   if (i % 2 == 0)
    //   {
    //     r *= 2;
    //   }
    //   else
    //   {
    //     r *= 4;
    //   }
    // }
    res += r;
  }

  return  height * M_PI * res;
}

int main()
{
  int N;
  double V;

  cin >> V;
  cin >> N;

  int min = 0;
  double min_diff = numeric_limits<double>::max();

  for (int i = 0; i < N; i++)
  {
    double a, b, h;

    cin >> a;
    cin >> b;
    cin >> h;

    double cv = vol(a, b, h);
    // cout << cv << endl;

    double diff = abs(cv - V);

    if (diff < min_diff)
    {
      min_diff = diff;
      min = i;
    }
  }

  cout << min << endl;

  return 0;
}

File: flowers.py
import math


# def incomplete_gamma(p, x):
#     return 4 * x / h(p, x)


# def h(p, x):
#     return 3 * (x ** 2 - p) + math.sqrt((x ** 2 - p) ** 2 + 8 * (x ** 2 + p))


# def vol(a, b, h):
#     return math.pi * (0.25) * (
#         math.sqrt(2 * math.pi) * (a ** 2) * math.erf(math.sqrt(2) * h) -
#         (4 * a * b * math.pow(h ** 2, 1/4) * incomplete_gamma(3/4, h ** 2)) / math.sqrt(h) +
#         2 * (b ** 2) * (h ** 2)
#     )

# def vol(a, b, h):
#     return (0.5) * math.pow(math.pi, 3/2) * a * math.erf(h) + (b ** 2) * (h ** 2) / 2

# print(vol(1,2, 2))


# N = 2
# V = 25
# h = 2

# graduations = 200

# height = h / graduations

# for i in range()

# print()
def f(a, b, x):
    return a * math.pow(math.e, -x ** 2) + b * math.sqrt(x)


def vol(a, b, h):
    grads = 1000000
    height = h / grads

    res = 0

    for i in range(grads):
        x = i * height
        r =  f(a, b, x)
        res += height * math.pi * r ** 2

    return res


[V, N] = [x for x in input().strip().split(" ")]

V = float(V)
N = int(N)

for i in range(N):
    [a, b, h] = [float(x) for x in input().strip().split(" ")]

    print(vol(a, b, h))
